gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/resources/endpoint/authorization/AuthorizationEndpoint.java

```diff
@@ -150,7 +150,7 @@ private void cleanSession(RoutingContext context) {
         context.session().remove(ConstantKeys.PASSWORDLESS_CHALLENGE_KEY);
         context.session().remove(ConstantKeys.PASSWORDLESS_CHALLENGE_USERNAME_KEY);
         context.session().remove(ConstantKeys.MFA_ENROLLMENT_COMPLETED_KEY);
-        context.session().remove(ConstantKeys.MFA_CHALLENGE_COMPLETED_KEY);
```

Comment:
**question**: don't think we should remove this line. This session attribute is IMO only relevant during the ongoing auth flow. Once authenticated, only the STRONGLY_AUTH attribute should remains in the session to be able to test the stepUp. wdy?

Reply:
if it is removed here, the session has no information that mfa challenge was completed before for example when rules are satisfied (on diagram: is valid session)

Reply:
> if it is removed here, the session has no information that mfa challenge was completed before for example when rules are satisfied (on diagram: is valid session)

🤔 ok, let go with it, I will rethink about it when all MFA rework will be done to have full picture